### PR TITLE
[Model] Expose size to Idefics3 as mm_processor_kwargs

### PR DESCRIPTION
**Pull request by Isotr0py**
- Expose `size` to initialize `Idefics3Processor`/`Idefics3ImageProcessor`, which can be used to reduce resized image size and save memory: https://huggingface.co/HuggingFaceM4/Idefics3-8B-Llama3#model-optimizations

cc @jeejeelee @B-201

### CODE REVIEW

File: vllm/model_executor/models/idefics3.py [hunk 1]
```diff
@@ -72,16 +71,38 @@ class Idefics3ImageEmbeddingInputs(TypedDict):
     """
 
 
+class Idefics3ProcessorSize(NamedTuple):
```
Comment (DarkLight1337) [inline vllm/model_executor/models/idefics3.py:74]: I think a `TypedDict` would be more appropriate for this?

File: vllm/model_executor/models/idefics3.py [hunk 2]
```diff
@@ -14,15 +14,14 @@
...
+from typing import (Dict, Iterable, List, Literal, Mapping, NamedTuple,
+                    Optional, Tuple, TypedDict, Union)
 
 import torch
 import torch.utils.checkpoint
 from PIL import Image
 from torch import nn
-# Temporary solution for transformers below 4.46.0.
-from transformers import PretrainedConfig as Idefics3Config
+from transformers import Idefics3Config, Idefics3ImageProcessor
```
Comment (DarkLight1337) [inline vllm/model_executor/models/idefics3.py:24]: We should not change this until upgrading the dependency versions.

Comment (Isotr0py) [reply vllm/model_executor/models/idefics3.py:24]: Oh, I forgot this.

File: vllm/model_executor/models/idefics3.py (hunk 1 above)
Comment (Isotr0py) [reply vllm/model_executor/models/idefics3.py:76]: `TypeDict` is unhashable and will raise `TypeError: unhashable type: 'dict'` when using with `lru_cache`.

However, `NamedTuple` is hashable, and we just need to override the `__getitem__` and `__contains__` to make it work like a dict.

Comment (DarkLight1337) [reply vllm/model_executor/models/idefics3.py:76]: I see, thanks for the explanation!

Comment (DarkLight1337) [reply vllm/model_executor/models/idefics3.py:76]: Can you add a note in the code for this?

Comment (Isotr0py) [reply vllm/model_executor/models/idefics3.py:76]: Added.

File: examples/offline_inference_vision_language.py [hunk 3]
```diff
@@ -385,7 +385,12 @@ def run_idefics3(question: str, modality: str):
     llm = LLM(model=model_name,
               max_model_len=8192,
               max_num_seqs=2,
-              enforce_eager=True)
+              enforce_eager=True,
+              mm_processor_kwargs={
```
Comment (jeejeelee) [inline examples/offline_inference_vision_language.py:389]: Should we add comments to explain why we need to add this parameter?

**Review by DarkLight1337 — APPROVED** [review]

File: examples/offline_inference_vision_language.py (hunk 3 above)
Comment (Isotr0py) [reply examples/offline_inference_vision_language.py]: Sure.

**Comment by jeejeelee** [comment]
This is unrelated to this PR, but I feel the test script names in this [directory](https://github.com/vllm-project/vllm/tree/main/tests/models/decoder_only/vision_language/mm_processor_kwargs) are a bit unreasonable.

**Comment by Isotr0py** [comment]
Perhaps we can consolidate the `mm_processor_kwargs` as well?

**Comment by DarkLight1337** [comment]
Let's do that in a separate PR.

**Comment by jeejeelee** [comment]
> Perhaps we can consolidate the `mm_processor_kwargs` tests as well?

Good idea. Perhaps https://github.com/vllm-project/vllm/tree/main/tests/multimodal is a better choice

**Pull request merged**